### ISSUE TITLE
rSync repo failing to clone code (Permission Denied (public key))

### ISSUE THREAD

**Issue by ryan-kimber** [Shippable/support#3840]
Hi guys,

All of a sudden this afternoon, one of our rSync repos started failing when we committed changes to our config (or tried to manually execute it).

I've tried the following things, but it is still failing:

- I made sure the SSH key in the settings on Shippable is also set up on the GitHub repository as deploy credentials with full access
- I tried updating our GitHub token and updated the GitHub SCM settings in Shippable
- I tried adding our 'continuous-deployment' repo to the SPOG so that I could go into it's settings and do a reset, as described in [this ticket](https://github.com/Shippable/support/issues/3523)
- I checked to see if GitHub is having any issues, but their status page is clean, and my other resources work, just not the rSync one that defines all the project resources (the repo that contains my resource, jobs, triggers, etc).

I am pretty stuck until I can get this fixed. Thanks for your help!

[BUILD LOG](https://app.shippable.com/github/formhero/jobs/continuous-deployment_develop_rSync/builds/59bf13626aa85a070024a1e0/console)

**Comment by trriplejay** [comment]
Hey Ryan,

the 'reset' instructions only work for resetting the key used for CI.  Currently we don't have a way to "reset" a syncRepo. I'll add that feature to our backlog.

In the meantime, I will email you the public key that needs to be in your syncRepo's "deploy keys" section on the github repo settings page.

Do you think it's possible that this key was deleted from your repo around the time that this started to fail?

**Comment by ryan-kimber** [comment]
Thanks for your help John.

It's possible it was deleted several weeks ago, but without any changes to that repo, it hadn't run again.

Adding the key you e-mailed me worked. That key was different than the one I can see in my settings though. Is there nowhere to see the deploy key for an rSync repo?

**Comment by trriplejay** [comment]
no, unfortunately. I've opened an internal ticket to address that, though. It's likely that we will add the ability to reset from the repo's resource page.
